R/uniDimFrequentistOutput.R

```diff
@@ -0,0 +1,196 @@
+  # for (i in 1:5) {
+  #   if (!derivedOptions[["selectedEstimators"]][i]) {
+  #     derivedOptions[["itemDroppedSelected"]][i] <- derivedOptions[["selectedEstimators"]][i]
+  #   }
+  # }
+
+  itemTable <- createJaspTable(gettext("Frequentist Individual Item Reliability Statistics"))
+  itemTable$dependOn(options = c("omegaItem", "alphaItem", "lambda2Item", "lambda6Item", "glbItem",
+                                 "itemMean", "itemRestCor", "itemSd",
+                                 "omegaScale", "alphaScale", "lambda2Scale", "lambda6Scale", "glbScale"))
+  # adding the scale options fixes a bug, where the item table would remain displayed
+  # after one had checked a scale coefficient box and the item coefficient box and then unchecked the scale coeff box
+
+  itemTable$addColumnInfo(name = "variable", title = gettext("Item"), type = "string")
+
+  itemDroppedSelected <- derivedOptions[["itemDroppedSelected"]]
+  coefficientsTable <- derivedOptions[["namesEstimators"]][["tables_item"]]
+  overTitle <- gettext("If item dropped")
+
+  idxSelected <- which(itemDroppedSelected)
+  coefficients <- derivedOptions[["namesEstimators"]][["coefficients"]]
+  idMatchedNames <- which(!is.na(charmatch(names(model), names(itemDroppedSelected[idxSelected]))))
+
+  for (i in idxSelected) {
+    if (coefficientsTable[i] %in% coefficients) {
+      itemTable$addColumnInfo(name = paste0("pointEst", i), title = coefficientsTable[i], type = "number",
+                              overtitle = overTitle)
+    } else {
+      itemTable$addColumnInfo(name = paste0("pointEst", i), title = coefficientsTable[i], type = "number")
+    }
+  }
+
+  if (is.null(model[["empty"]])) {
+    tb <- data.frame(variable = model[["itemsDropped"]])
+    z <- 1
+    for (i in idxSelected) {
+      newtb <- cbind(pointEst = model[[idMatchedNames[z]]][["itemDropped"]])
+      colnames(newtb) <- paste0(colnames(newtb), i)
+      tb <- cbind(tb, newtb)
+      z <- z+1
+    }
+    itemTable$setData(tb)
+
+    if (!is.null(unlist(options[["reverseScaledItems"]]))) {
+      itemTable$addFootnote(sprintf(ngettext(length(options[["reverseScaledItems"]]),
+                                             "The following item was reverse scaled: %s. ",
+                                             "The following items were reverse scaled: %s. "),
+                                    paste(options[["reverseScaledItems"]], collapse = ", ")))
+    }
+
+  } else if (length(model[["itemsDropped"]]) > 0) {
+    itemTable[["variables"]] <- model[["itemsDropped"]]
+
+    if (!is.null(unlist(options[["reverseScaledItems"]]))) {
+      itemTable$addFootnote(sprintf(ngettext(length(options[["reverseScaledItems"]]),
+                                             "The following item was reverse scaled: %s. ",
+                                             "The following items were reverse scaled: %s. "),
+                                    paste(options[["reverseScaledItems"]], collapse = ", ")))
+    }
```

Comment:
could you create a function for
```r
sprintf(ngettext(length(options[["reverseScaledItems"]]),
                                             "The following item was reverse scaled: %s. ",
                                             "The following items were reverse scaled: %s. "),
                                    paste(options[["reverseScaledItems"]], collapse = ", "))
```
?

That way the string is only translated once.